pilot/pkg/model/validation.go

```diff
@@ -214,7 +214,10 @@ func (l Labels) Validate() error {
 
 // ValidateFQDN checks a fully-qualified domain name
 func ValidateFQDN(fqdn string) error {
-	return appendErrors(checkDNS1123Preconditions(fqdn), validateDNS1123Labels(fqdn))
+	if err := checkDNS1123Preconditions(fqdn); err != nil {
+		return err
```

Comment:
you need to appendErrors(..) because we retain the error stack

Reply:
You only need appendErrors here if it proceeds to call validateDNS1123Labels, but I don't see any value in doing further checking if the name is already known to be invalid.